runtime/src/main/kotlin/kotlin/Array.kt

```diff
@@ -17,6 +17,7 @@ import kotlin.native.internal.PointsTo
  * for more information on arrays.
  */
 @ExportTypeInfo("theArrayTypeInfo")
+@ShareByValue
```

Comment:
Array classes don't have custom `equals`/`hashCode`, so this doesn't seem correct according to our discussion.